cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -212,112 +211,92 @@ public TableKey apply(Table table) throws Exception {
-                }
-              });
-            } catch (Exception e) {
-              if (e instanceof NamespaceNotFoundException) {
-                LOG.warn("Namespace {} does not exist. Only delete metadata for it", namespaceId.getEntityName(), e);
-              } else {
-                LOG.warn("Error while accessing namespace {} for log cleanup, skipping it",
-                         namespaceId.getEntityName(), e);
-                continue;
-              }
-            }
 
             for (final Map.Entry<byte[], byte[]> entry : row.getColumns().entrySet()) {
               try {
                 byte[] colName = entry.getKey();
                 URI file = new URI(Bytes.toString(entry.getValue()));
+
                 if (LOG.isDebugEnabled()) {
                   LOG.debug("Got file {} with start time {}", file, Bytes.toLong(colName));
                 }
+                URI fileUri = new URI(Bytes.toString(entry.getValue()));
 
-                if (Strings.isNullOrEmpty(namespacedLogDir)) {
-                  LOG.warn("File {} is not present and will be deleted from metadata because namespace {} " +
-                             "does not exist", Bytes.toString(entry.getValue()), namespaceId.getEntityName());
-                  table.delete(rowKey, colName);
-                  deletedColumns++;
-                  continue;
-                }
-
-                Location fileLocation = impersonator.doAs(namespaceId, new Callable<Location>() {
-                  @Override
-                  public Location call() throws Exception {
-                    return rootLocationFactory.create(new URI(Bytes.toString(entry.getValue())));
-                  }
-                });
-
-                if (!fileLocation.exists()) {
-                  LOG.warn("Log file {} does not exist, but metadata is present", file);
-                  table.delete(rowKey, colName);
-                  deletedColumns++;
-                } else if (fileLocation.lastModified() < tillTime) {
-                  // Delete if file last modified time is less than tillTime
-                  callback.handle(namespaceId, fileLocation, namespacedLogDir);
+                if (filesToRemove.containsKey(fileUri)) {
                   table.delete(rowKey, colName);
-                  deletedColumns++;
+                  count++;
                 }
               } catch (Exception e) {
-                if (e instanceof NamespaceNotFoundException) {
-                  LOG.warn("File {} is not present because namespace {} does not exist",
-                           Bytes.toString(entry.getValue()), namespaceId.getEntityName());
-                } else {
-                  LOG.error("Got exception deleting file {}", Bytes.toString(entry.getValue()), e);
-                }
+                LOG.error("Got exception deleting file {}, ignoring it.", Bytes.toString(entry.getValue()), e);
```

Comment:
I wonder whether this can still throw exceptions? If so, then not from deleting a file (because that is not done here any longer).